x/sp/types/message.go

```diff
@@ -194,14 +205,23 @@ func (msg *MsgEditStorageProvider) ValidateBasic() error {
 			return errors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid gc address (%s)", err)
 		}
 	}
-
 	if msg.BlsKey != "" {
+		if msg.BlsProof == "" {
+			return errors.Wrapf(gnfderrors.ErrInvalidBlsSignature, "bls proof is not provided")
+		}
 		blsPk, err := hex.DecodeString(msg.BlsKey)
 		if err != nil || len(blsPk) != sdk.BLSPubKeyLength {
 			return errors.Wrapf(sdkerrors.ErrInvalidPubKey, "invalid bls pub key")
 		}
+		blsProof, err := hex.DecodeString(msg.BlsProof)
+		if err != nil || len(blsProof) != sdk.BLSSignatureLength {
+			return errors.Wrapf(sdkerrors.ErrorInvalidSigner, "invalid bls signature")
+		}
+		_, err = bls.SignatureFromBytes(blsProof)
```

Comment:
check this in MsgCreateStorageProvider as well

Reply:
Added